xskillscore/core/probabilistic.py

```diff
@@ -485,3 +489,123 @@ def discrimination(
+            If False (default), the new object will be returned without attributes.
+
+        Returns
+        -------
+        xarray.Dataset or xarray.DataArray
+            The relative frequency of occurrence for each probability bin
+        xarray.Dataset or xarray.DataArray
+            The sample size of each probability bin
+
+        Examples
+        --------
+        >>> forecasts = xr.DataArray(np.random.normal(size=(3,3,3)),
+        ...                          coords=[('x', np.arange(3)),
+        ...                                  ('y', np.arange(3)),
+        ...                                  ('ensemble', np.arange(3))])
+        >>> observations = xr.DataArray(np.random.normal(size=(3,3)),
+        ...                            coords=[('x', np.arange(3)),
+        ...                                    ('y', np.arange(3))])
+        >>> rel, samples = reliability(observations > 0.1, (forecasts > 0.1).mean('ensemble'), dim='x')
+
+        Notes
+        -----
+        See http://www.cawcr.gov.au/projects/verification/
+    """
+
+    def _reliability(o, f, bin_edges):
+        """Return the reliability and number of samples per bin
+        """
+        # I couldn't get dask='parallelized' working in this case so dealing with dask arrays explicitly
+        is_dask_array = isinstance(o, darray.core.Array) | isinstance(
+            f, darray.core.Array
+        )
+
+        if is_dask_array:
+            r = []
+            N = []
+        else:
+            r = np.zeros((*o.shape[:-1], len(bin_edges) - 1), dtype=float)
+            N = np.zeros_like(r)
+
+        for i in range(len(bin_edges) - 1):
+            # Follow xhistogram: all bins are half-open
+            # https://github.com/xgcm/xhistogram/issues/18
+            f_in_bin = (f >= bin_edges[i]) & (f < bin_edges[i + 1])
+            o_f_in_bin = o & f_in_bin
+            N_f_in_bin = f_in_bin.sum(axis=-1)
+            N_o_f_in_bin = o_f_in_bin.sum(axis=-1)
+            if is_dask_array:
+                r.append(N_o_f_in_bin / N_f_in_bin)
+                N.append(N_f_in_bin)
+            else:
+                r[..., i] = N_o_f_in_bin / N_f_in_bin
+                N[..., i] = N_f_in_bin
+
+        if is_dask_array:
+            return (
+                darray.stack(r, axis=-1).rechunk({-1: -1}),
+                darray.stack(N, axis=-1).rechunk({-1: -1}),
+            )
+        else:
+            return r, N
+
+    _fail_if_dim_empty(dim)
+
+    # Compute over all dims if dim is None
+    if dim is None:
+        dim = list(observations.dims)
+
+    dim, _ = _preprocess_dims(dim, observations)
+    observations, forecasts, stack_dim, _ = _stack_input_if_needed(
+        observations, forecasts, dim, weights=None
+    )
+
+    rel, samp = xr.apply_ufunc(
+        _reliability,
+        observations,
+        forecasts,
+        probability_bin_edges,
+        input_core_dims=[[stack_dim], [stack_dim], []],
+        dask='allowed',
```

Comment:
```suggestion
        dask='parallelized',
```

Reply:
ah ok now I see
`a,b=xs.reliability(obs3>.5,(fct3>.5).mean('member').chunk())` doesnt work. in the new unreleased xarray version they allow multiple outputs for `dask='parallelized'`.

Reply:
Even when I changed `_reliability` to only have one output, this still wouldn't work. It worked fine with `numpy` arrays but would fail with some sort of chunking error when using a `dask` array + `dask='parallelized'`

Reply:
Maybe we can revisit this once the new version of `xarray` is released. Regardless, I think it should be faster to deal with `dask` arrays explicitly with `dask='allowed'` rather than using `dask='parallelized'` on a `numpy` function (http://xarray.pydata.org/en/stable/dask.html)